objectRecognitionSource/src/main/java/edu/ml/tensorflow/util/ImageUtil.java

```diff
@@ -63,7 +64,33 @@ public void saveImage(final BufferedImage image, final String target) {
         if (outputDir != null) {
             try {
                 IOUtil.createDirIfNotExists(new File(outputDir));
-                ImageIO.write(image,"jpg", new File(outputDir + File.separator + target));
+                if (longEdge == 0) {
+                    ImageIO.write(image, "jpg", new File(outputDir + File.separator + target));
+                } else {
+                    int width = image.getWidth();
+                    int height = image.getHeight();
+                    if (longEdge > Math.max(width, height)) {
+                        LOGGER.error("The longEdge value is too large, resizing cannot enlarge the original image. "
+                                + "Original image resolution will not be changed");
+                        ImageIO.write(image, "jpg", new File(outputDir + File.separator + target));
+                        longEdge = 0;
```

Comment:
This unsets longEdge for everything. Couldn't some source send back various images with different dimensions?  Seems like we just want to skip the resize, not change the expected behavior.

Reply:
Just talked about this, forget the comment above :).